### PR TITLE
Add intercom to order app

### PR DESCRIPTION
**Pull request by ds300**
Following artsy/force#4515 we're now invoking the setup logic for Intercom.

Didn't have a chance to test this on eigen because of the MP issues that prevented our QA session today, so give it a go before merging if you have the time. (note you'll need to grab the `INTERCOM_*` env vars from force staging then run force locally with this reaction branch linked. And then you need to configure eigen to use local force (think this can be done from the dev menu, but I do it by editing `ARDefaults.m`)

### CODE REVIEW

File: src/Apps/Order/redirects.tsx [hunk 1]
```diff
@@ -41,7 +40,7 @@ export const confirmRouteExit = (
 }
 
 const goToStatusIf = (
-  pred: (order: routes_OrderQueryResponse["order"]) => boolean,
+  pred: (order: redirects_order) => boolean,
```
Comment (sweir27) [inline src/Apps/Order/redirects.tsx]: Was this just leftover from before?

Comment (ds300) [reply src/Apps/Order/redirects.tsx]: Yep

**Comment by ds300** [comment]
I checked on eigen and it works beautifully. Intercom is very impressive!

**Comment by artsyit** [comment]
:rocket: PR was released in v17.11.6 :rocket:

**Pull request merged**